File: gitdot-core/src/dto/domain/ci/task.rs
mod update_task;

use chrono::{DateTime, Utc};
use uuid::Uuid;

use crate::model::{Task, TaskStatus};

pub use update_task::UpdateTaskRequest;

#[derive(Debug, Clone)]
pub struct TaskResponse {
    pub id: Uuid,
    pub repository_id: Uuid,
    pub build_id: Uuid,
    pub s2_uri: String,
    pub name: String,
    pub command: String,
    pub status: TaskStatus,
    pub waits_for: Vec<Uuid>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

impl From<Task> for TaskResponse {
    fn from(task: Task) -> Self {
        Self {
            id: task.id,
            repository_id: task.repository_id,
            build_id: task.build_id,
            s2_uri: task.s2_uri,
            name: task.name,
            command: task.command,
            status: task.status,
            waits_for: task.waits_for,
            created_at: task.created_at,
            updated_at: task.updated_at,
        }
    }
}

File: gitdot-web/packages/gitdot-api/src/resource/task.ts
import { z } from "zod";

export const TaskStatus = z.enum([
  "blocked",
  "pending",
  "assigned",
  "running",
  "success",
  "failure",
]);
export type TaskStatus = z.infer<typeof TaskStatus>;

export const TaskResource = z.object({
  id: z.uuid(),
  repository_id: z.uuid(),
  build_id: z.uuid(),
  s2_uri: z.string(),
  name: z.string(),
  command: z.string(),
  status: TaskStatus,
  waits_for: z.array(z.uuid()),
  created_at: z.iso.datetime(),
  updated_at: z.iso.datetime(),
});
export type TaskResource = z.infer<typeof TaskResource>;

export const PollTaskResource = z.object({
  id: z.uuid(),
  repository_id: z.uuid(),
  owner_name: z.string(),
  repository_name: z.string(),
  s2_uri: z.string(),
  name: z.string(),
  command: z.string(),
  status: TaskStatus,
});
export type PollTaskResource = z.infer<typeof PollTaskResource>;

export const TaskTokenResource = z.object({ token: z.string() });
export type TaskTokenResource = z.infer<typeof TaskTokenResource>;
